x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/rate/InternalResetTrackingRateTests.java

```diff
@@ -35,24 +36,50 @@ protected SearchPlugin registerPlugin() {
 
     @Override
     protected InternalResetTrackingRate createTestInstance(String name, Map<String, Object> metadata) {
-        return new InternalResetTrackingRate(name, null, metadata, 0, 0, 0, 0, 0);
+        return new InternalResetTrackingRate(name, null, metadata, 0, 0, 0, 0, 0, Rounding.DateTimeUnit.SECOND_OF_MINUTE);
     }
 
-    private static InternalResetTrackingRate rate(double startValue, double endValue, long startTime, long endTime, double resetComp) {
-        return new InternalResetTrackingRate("n", null, null, startValue, endValue, startTime, endTime, resetComp);
+    private static InternalResetTrackingRate rate(
+        double startValue,
+        double endValue,
+        long startTime,
+        long endTime,
+        double resetComp,
+        Rounding.DateTimeUnit rateUnit
+    ) {
+        return new InternalResetTrackingRate("n", null, null, startValue, endValue, startTime, endTime, resetComp, rateUnit);
     }
 
-    public void testReduction() {
-        List<InternalAggregation> rates = List.of(
-            rate(0, 10, 1000, 2000, 0),
-            rate(10, 20, 2000, 3000, 0),
-            rate(20, 5, 3000, 4000, 25), // internal reset
-            rate(5, 15, 4000, 5000, 0),
-            rate(0, 10, 5000, 6000, 0)  // cross-boundary reset
-        );
-        InternalAggregation reduced = rates.get(0).reduce(rates, null);
-        assertThat(reduced, instanceOf(Rate.class));
-        assertThat(((Rate) reduced).getValue(), equalTo(0.01));
+    public void testReductionSecond() {
+        testReduction(Rounding.DateTimeUnit.SECOND_OF_MINUTE, 0.01);
```

Comment:
This is the existing test returning 0.01 when unit = second (the default).

All other tests just check the result multiplying by 60, 60, 24 and so on...